environment.py

```diff
@@ -410,6 +411,15 @@ def env() -> Mapping[str, Optional[str]]:
         # https://cloud.google.com/bigquery/docs/locations
         'AZUL_TDR_SOURCE_LOCATION': None,
 
+        # The full set of BigQuery dataset locations of the TDR snapshots
+        # indexed across all deployments. The value of
+        # ``AZUL_TDR_SOURCE_LOCATION`` must always be an element of this set.
+        #
+        'AZUL_TDR_ALLOWED_SOURCE_LOCATIONS': json.dumps([
+            'US',  # Used by the prod deployment
+            'us-central1'  # Used by the dev and sandbox deployments
```

Comment:
Personally, I think the comments are a tad redundant but I'll defer to your expertise.